src/scripts.js

```diff
@@ -10,204 +10,162 @@
-//       <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
-//       <img id='${recipe.id}' tabindex='0' class='card-picture'
-//       src='${recipe.image}' alt='Food from recipe'>
-//       </div>`)
-//     })
-//   }
-// }
-
-function greetUser(user) {
-  const userName = document.querySelector('.user-name');
-  if(userName === null){
-    return 
-  }
-  userName.innerText += ' ' + user.name.split(' ')[0] + ' ' + user.name.split(' ')[1][0];
+function grabRecipes() {
+  fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/recipes/recipeData')
+  .then(response => response.json())
+  .then(recipeObject => {
+    recipeData = recipeObject.recipeData
+      domUpdates.displayAllRecipes(recipeData,favorites);
+  })
+  .catch(err => {
+    console.log(err);
+    alert('Sorry, the recipes failed to load. Try again later.')})
 }
 
-function displayPantry(pantry) {
-  if(pantryArea === null){
-    return 
-  }
-  pantryArea.innerHTML = ''
-  pantry.forEach(ingInPantry => {
-    pantryArea.innerHTML += `<article class="single-ingredient-card">
-  <article class="ingredient-name-area">
-    <h2 class="ingredient-name">${ingInPantry.name}</h2>
-  </article>
-  <article class="ingredient-quantity-section">
-    <button>-</button>
-    <p class = 'ingredient-quantity'>${ingInPantry.amount}</p>
-    <button>+</button>
-  </article>
-</article>`
+function grabUsers() {
+  return fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/users/wcUsersData',)
+  .then(response => {
+    return response.json()
+  })
+  .then(data => {
+    console.log(domUpdates.randomNumber)
+    domUpdates.greetUser(data.wcUsersData[domUpdates.randomNumber -1])
+     return data.wcUsersData;
+      })
+  .catch(err => {
+      console.log(err);
+      alert('Sorry, the user information failed to load. Try again later.');
+  })
+}
+  function grabIngredients() {
+  return fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/ingredients/ingredientsData')
+  .then(response => {
+      return response.json()
+  }).then(ingredientData => {
+      return ingredientData.ingredientsData;
+      })
+  .catch(err => {
+      console.log(err);
+      alert('Sorry, the ingredients failed to load. Try again later.');    
+})
+}
+function mergeFetchTimelines() {
+ Promise.all([grabUsers(), grabIngredients()])
+ .then(values => {
+   console.log(values[1])
+   let usersData = values[0]
+   ingredientsData = values[1]
+   let currentUser = createUser(usersData)
+   let pantry = createPantry(currentUser,ingredientsData)
+      domUpdates.displayPantry(pantry)
+ })
+  .catch(err => {
+      console.log(err);
+      alert('Sorry, the information failed to load. Try again later.');
+  })
+  
+}
+function findFavorites(ids,recipeData) {
+ 
+  favoriteRecipies = recipeData.filter(recipe =>{
+    return ids.includes(String(recipe.id))
+  })
+  console.log(favoriteRecipies)
+  return favoriteRecipies
+}
+function filterInputs(letters,ingredientsData){
```

Comment:
Could you explain how this function works?  It's a little confusing with all the nested filtering/finding

Reply:
I think that we are trying to find the recipe ingredient that matches the ingredient from the fetch request...but not sure what is happening in a 'line-by-line' way